### PR TITLE
Fixes resetting date picker's configuration

### PR DESCRIPTION
**Pull request by mrpiotr-dev**
### Context
Our usage of `pikadayInstance.config(newConfig)` pollutes config for everyone who wants to use Pikaday later (us also). Due to that fact, there is no option to use different configurations in `datePickerConfig` among date cells.

This PR changes the way we use Pikaday. Instead of reusing a single date picker instance per Handsontable, we will create and destroy Pikaday on open/close actions on the editor. `new Pikaday(config)` works stable and do not pollute next instances.

This PR DOES NOT cleanup code nor refactor the editor - I got information that there are plans to remove the third-party library from the built-in DateEditor.

### How has this been tested?
A) Use `steps to reproduce` mentioned in the root issue (#6636)
B) Run E2E tests

### Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)

### Related issue(s):
1. #6636

### Affected project(s):
- [x] `handsontable`

### CODE REVIEW

File: src/editors/dateEditor/dateEditor.js [hunk 1]
```diff
@@ -194,17 +191,6 @@ export class DateEditor extends TextEditor {
...
     }
-
-    this.datePickerStyle.display = 'block';
-    this.$datePicker.show();
-  }
-
-  /**
-   * Hide data picker.
-   */
-  hideDatepicker() {
```
Comment (budnix) [inline src/editors/dateEditor/dateEditor.js:205]: Can I ask you to update the TS definition file (https://github.com/handsontable/handsontable/blob/feature/issue-6636/handsontable.d.ts#L392)? I guess that removing the `hideDatepicker` method from there is enough.

**Review by budnix — CHANGES_REQUESTED** [review]
LGTM 👍  with one feedback from my side :)

**Comment by budnix** [comment]
> LGTM 👍 with one feedback from my side :)

...aaaand I think that after the changes one test does not pass. The `DateEditor should close calendar after picking new date` fails on CI and on my PC.

File: src/editors/dateEditor/dateEditor.js (hunk 1 above)
Comment (mrpiotr-dev) [reply src/editors/dateEditor/dateEditor.js:205]: Thank you for pointing unnecessary definitions out. I also removed the `types` test for `hideDatepicker()`.

**Comment by mrpiotr-dev** [comment]
> > LGTM 👍 with one feedback from my side :)
> 
> ...aaaand I think that after the changes one test does not pass. The `DateEditor should close calendar after picking new date` fails on CI and on my PC.

Yeah, I noticed that too. Unfortunately, locally on my PC works well, but on a laptop, it fails... 🙈 .
I have to verify what's going on. Maybe it's caused by diff OS, chrome's version, CPU load or... increased number of specs' dots and screen size 😂

**Comment by mrpiotr-dev** [comment]
The problem lies in differences in building `E2ERunner.html` for production and ci/dev env.
a) `test-e2e.js` uses `../dist/handsontable.css`, which doesn't contain pikaday's part (see TL;DR below)
b) `test-production.js` uses `../dist/handsontable.full.min.css`, which contains all necessary styles

TL;DR
Pikaday's `hide()` method adds `is-hidden` class name. To hide Pikaday's element, we have to include Pikaday's CSS file in a bundle. Our `development` config uses `empty-loader` to exclude Pikaday's CSS from `handsontable.css`.
I didn't have errors on Windows because that `empty-loader` doesn't work on Windows. Webpack doesn't exclude Pikaday's CSS from the `handsontable.css` file. So we have another problem with inconsistencies between platforms.
Unfortunately, because of that fact, I have to revert removed `hideDatepicker()`.

**Review by budnix — APPROVED** [review]
